topic4.user/UserServiceTest.java

```diff
@@ -0,0 +1,30 @@
+package bootcamp412.topic4_user;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class UserServiceTest {
+
+	@Test
+	public void test() {
+		// create
+		// read
+		Service pService = new ProxyService();
```

Comment:
[High] You have to use a ServiceFactory to return a proxy instance. The factory and client classes will hide the object construction complexity.construction.
So, you should have:
 UserService --> the interface
 UserServiceImpl --> the implementation 
 UserServiceProxy --> the proxy
 UserServiceLocalClient --> the client that construct/inject/use the implementation part
 UserServiceFactory --> creates the proxy and pass it the client in the construction